vantiq-cloud-infra-operations/docs/eng/kubernetes-upgrade.md

```diff
@@ -0,0 +1,206 @@
+# Kubernetes upgrade
+ This article describes how to upgrade a Kubernetes cluster running Vantiq by an infrastructure administrator in a Vantiq Private Cloud configuration, and what to keep in mind.  
+
+## AKS upgrade<a id="aks_upgrade"></a>
+
+There are two ways to upgrade AKS.  
+Method | Content | Requirements
+--|---|--
+Upgrade with Azure Portal  | Upgrade manually via Azure Portal in the order of Control Plane, Node Pools. | <ul><li>The execute user must have the necessary update permissions for the subscription.</li></ul>
+Upgrade with Terraform  | Update the version within the Terraform configuration file and apply it to Azure.  |<ul><li>The user of the Terraform execution environment must have the necessary update permissions for the subscription.</li><li>There must be no deviation in the configuration of Terraform before the update is applied (No configuration changes have been made outside of Terraform after configuring with Terraform).</li></ul>
+
+#### Upgrade with Azure Portal  
+1. Upgrade Control Plane    
+   There are two options **to upgrade only the Control Plane** or **to also upgrade all node pools**, **do the former**. For Nodepool, the dependencies of the Vantiq pod group should be considered and executed in order.  
+
+   ![aks1](../../imgs/kubernetes-upgrade/aks-1.png)
+
+1. Upgrade Nodepool   
+   Select the target Nodepools from the Node pools and upgrade them respectively. Keep in mind the following points;  
+
+    - Nodepools running Vantiq pod and Mongodb pod should be upgraded at different times from other Nodepools. It is acceptable to upgrade them at the same time if the criteria are met.  
+    - When upgrading Nodepool running Vantiq pod, confirm that both Mongodb pod and Keycloak pod are running normally (status=Running) before upgrading.  
+
+    ![aks2](../../imgs/kubernetes-upgrade/aks-2.png)  
+
+    Select Nodepool, execute **Update Kubernetes**.  
+
+    ![aks3](../../imgs/kubernetes-upgrade/aks-3.png)  
+
+
+#### Upgrade with Terraform
+
+The following is the procedure assuming that it was installed using the template ["Azure AKS for Vantiq with Terraform"](../../../vantiq-cloud-infra-operations/terraform_azure/readme_en.md). If it used a different template, make the necessary changes accordingly.  
```

Comment:
```suggestion
The following is the procedure assuming that it was installed using the template ["Azure AKS for Vantiq with Terraform"](../../../vantiq-cloud-infra-operations/terraform_azure/readme_en.md). In case that different template was used, make the changes as per the template.  
```